injector.js

```diff
@@ -96,9 +96,9 @@ let recompile = async function(metadata, sources) {
 let findMetadataFile = function(files) {
   for (let i in files) {
     try {
-      let m = JSON.parse(files[i])
-      if (m['language'] === 'Solidity') {
-        return m
+      const file = JSON.parse(JSON.parse(files[i]))
```

Comment:
Is this double parse required?

Reply:
In my testing, unfortunately yes, but that was an item to revisit. Presumably that can get cleaned up further up the stack.

Reply:
@cgewecke From what I'm seeing (using your example test files), metadata files come in "over-stringified", i.e., the first pass of JSON.parse() still returns a string. 

The existing, deployed proof of concept on verification.komputing.org appears to produce the same results when using these files; my tests produce the same `Metadata file not found` error. Possibly @chriseth has tested with files that have been generated via another method?

In any event, I reverted this change, but moved some conditional logic into `server.js` to parse the file if it's able. I believe there's no "harm" in double parsing, but its arguably cleaner now. Open to other suggestions.

Reply:
@marcgarreau  I think your solution is the right one... my understanding of this is that:

when the metadata is generated using solc's JSON-IO interface and saved to file as part of an artifact, it's stringified and looks like:

```
"{\"compiler\":{\"version\":\"0.6.0+commit.26b70077\"},\"langua....
```

If you load the artifact and query its type
```js
typeof compilerOutput.metadata
> string
```

Uploadeding the string to IPFS with `ipfs.add` will locate it at the same hash as the one encoded in deployedBytecode. The monitor uses it to find metadata for newly deployed contracts.

On the other hand, when metadata is generated at the cli with `solc --metadata -o metadata.json`, the file contents look like:
```
{"compiler":{"version":"0.6.0+commit.26b70077"},"langua...
```

The cli version is used to organize recompilation and compare its outcome with the deployedBytecode at the user supplied contract address. 

So it looks like there are two use cases and two formats - the second is normal for the UI but maybe it makes sense to support both versions there anyway?